source/Octopus.Client/UrlHelper.cs

```diff
@@ -0,0 +1,21 @@
+using Octopus.Client.Extensibility;
+
+namespace Octopus.Client
+{
+    internal static class UrlHelper
```

Comment:
A year ago I would have done it this way as well, but we've been burned by code that starts this way and are trying to limit statics to extension methods where possible.

Would there be value in making a `ResourceSelfLinkExtractor` class, who's sole job is to extract a link from a given resource? Then we just new that up as a property on the `OctopusClient` and use it multiple times.

E.g. pseudocode:
```
class OctopusClient {
   readonly ResourceSelfLinkExtractor resourceSelfLinkExtractor =  new ResourceSelfLinkExtractor();
   ...
   // Then inside whatever methods we need...
   var selfLink = resourceSelfLinkExtractor.GetSelfUrlOrNull(response.ResponseResource) ?? path;
   ...
}
```

It seems small, but we may get a lot more flexibility/control in the future if we ever wanted to tweak that during tests, and this stops static helper classes/methods like this spreading via copy/pasta.

Reply:
Looking back on it, I can see how this is much more maintainable. I've updated it as you've suggested in ab6490704779d9578decc92ecaa8f3e46a27407c